engine-tests/src/test/java/org/terasology/logic/characters/VisualCharacterSystemTest.java

```diff
@@ -0,0 +1,183 @@
+    private long nextEntityId = 1;
+
+    @Before
+    public void setup() throws Exception {
+        this.system = new VisualCharacterSystem();
+        Context context = new ContextImpl();
+
+        this.localPlayer = Mockito.mock(LocalPlayer.class);
+        context.put(LocalPlayer.class, localPlayer);
+        Mockito.doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                return clientEntityReturnedByLocalPlayer;
+            }
+        }).when(localPlayer).getClientEntity();
+
+        this.entityManager = Mockito.mock(EntityManager.class);
+
+        Mockito.doReturn(Mockito.mock(EntityBuilder.class)).when(entityManager).newBuilder();
+
+        context.put(EntityManager.class, this.entityManager);
+        InjectionHelper.inject(system, context);
+        system.setCreateAndAttachVisualEntityStrategy((entityBuilder, characterEntity) -> Mockito.mock(EntityRef.class));
+
+    }
+
+    private void recordEntityEventsToList(EntityRef entityRefMock, List<Event> listToFill) {
+        Mockito.when(entityRefMock.send(Mockito.any())).then(new Answer<Event>() {
+            @Override
+            public Event answer(InvocationOnMock invocation) throws Throwable {
+                Event event = invocation.getArgument(0);
+                listToFill.add(event);
+                return event;
+            }
+        });
+    }
+
+    /**
+     * This test verifies that:
+     * <ul>
+     *     <li>No visual character gets created (via event) for the own character (as it is first person)</li>
+     *     <li>That the system can deal with LocalPlayer and characters not being properly linked when
+     *     the character entities get loaded/created</li>
+     *     <li>A visual character gets created (via event) for charcters that were already present when the player
+     *     joined</li>
+     *     <li>A visual character gets created (via event) for characters that joins afterwards </li>
+     * </ul>
+     */
+    @Test
+    public void testSendingOfCreateVisualCharacterEvent() {
+        EntityRef clientEntity = mockEntityWithUniqueId();
+        EntityRef otherClientEntity = mockEntityWithUniqueId();
+
+        EntityRef ownCharacterEntity = mockEntityWithUniqueId();
+        List<Event> ownCharacterEntityEvents = new ArrayList<>();
+        recordEntityEventsToList(ownCharacterEntity, ownCharacterEntityEvents);
+
+        EntityRef otherCharacterEntity = mockEntityWithUniqueId();
+        List<Event> otherCharacterEntityEvents = new ArrayList<>();
+        recordEntityEventsToList(otherCharacterEntity, otherCharacterEntityEvents);
+
+        clientEntityReturnedByLocalPlayer = EntityRef.NULL;
+
+        /*
+         * Simulate activation before entity is done
+         * since the character is not properly linked yet nothing should happen
+         */
+        system.onActivatedVisualCharacter(OnActivatedComponent.newInstance(), otherCharacterEntity);
```

Comment:
Syntax error here and two further instance of the same situation - no `VisualCharacterComponent` is supplied to the method despite being required. I suspect the target method was changed after the test was written? Easy to miss since `engine-tests` won't force recompile when just trying to re-run the game for play testing.

This is what is marking the Jenkins build as broken. I tried just throwing in a `new VisualCharacterComponent()` in the three spots but then the test hit an similar sounding NPE so it probably needs a quick bit of tweaking